File: src/main/java/upbrella/be/store/dto/response/SingleClassificationResponse.java
package upbrella.be.store.dto.response;

import com.querydsl.core.annotations.QueryProjection;
import lombok.*;
import upbrella.be.store.entity.Classification;
import upbrella.be.store.entity.ClassificationType;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class SingleClassificationResponse {

    private long id;
    private ClassificationType type;
    private String name;
    private double latitude;
    private double longitude;

    @QueryProjection
    public SingleClassificationResponse(long id, ClassificationType type, String name, double latitude, double longitude) {
        this.id = id;
        this.type = type;
        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public static SingleClassificationResponse ofCreateClassification(Classification classification) {

        return SingleClassificationResponse.builder()
                .id(classification.getId())
                .type(classification.getType())
                .name(classification.getName())
                .latitude(classification.getLatitude())
                .longitude(classification.getLongitude())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleClassificationResponse.kt
package upbrella.be.store.dto.response

import com.querydsl.core.annotations.QueryProjection
import upbrella.be.store.entity.Classification
import upbrella.be.store.entity.ClassificationType

data class SingleClassificationResponse @QueryProjection constructor(
    val id: Long?,
    val type: ClassificationType?,
    val name: String?,
    val latitude: Double?,
    val longitude: Double?
) {
    companion object {

        fun ofCreateClassification(classification: Classification): SingleClassificationResponse {
            return SingleClassificationResponse(
                id = classification.id,
                type = classification.type,
                name = classification.name,
                latitude = classification.latitude,
                longitude = classification.longitude
            )
        }
    }
}
